cunumeric/module.py

```diff
@@ -5327,6 +5336,630 @@ def nancumsum(
     )
 
 
+def _get_non_nan_reduction_code_if_applicable(
+    dtype: npt.DTypeLike, unary_reduction_code: UnaryRedCode
+) -> UnaryRedCode:
+    """
+    Return the equivalent non-nan reduction op codes if the datatype of the
+    array doesn't belong to one of the allowed datatypes. Raise an error if the
+    datatype is disallowed, which is currently complex64 and complex128.
+    """
+
+    equivalent_non_nan_unary_red_ops: Dict[UnaryRedCode, UnaryRedCode] = {
+        UnaryRedCode.NANARGMAX: UnaryRedCode.ARGMAX,
+        UnaryRedCode.NANARGMIN: UnaryRedCode.ARGMIN,
+        UnaryRedCode.NANMAX: UnaryRedCode.MAX,
+        UnaryRedCode.NANMIN: UnaryRedCode.MIN,
+    }
+
+    assert unary_reduction_code in equivalent_non_nan_unary_red_ops
+
+    # raise error for disallowed datatypes
+    disallowed_dtypes: list[np.dtype[Any]] = [
+        np.dtype(np.complex64),
+        np.dtype(np.complex128),
+    ]
+    if dtype in disallowed_dtypes:
+        raise NotImplementedError(
+            "operation is not supported for complex64 and complex128 types"
+        )
+
+    # use non-NaN API if the datatype is not floating point type
+    allowed_dtypes: list[np.dtype[Any]] = [
+        np.dtype(np.float16),
+        np.dtype(np.float32),
+        np.dtype(np.float64),
+    ]
+    if dtype in allowed_dtypes:
+        reduction_code = unary_reduction_code
+    else:
+        reduction_code = equivalent_non_nan_unary_red_ops[unary_reduction_code]
+
+    return reduction_code
```

Comment:
```suggestion
    if dtype in allowed_dtypes:
        return unary_reduction_code
    else:
        return equivalent_non_nan_unary_red_ops[unary_reduction_code]
```
I'd actually do `return a if cond b`, but the variable name seems a bit long to do that.

Reply:
yeah, the variable and function names are a bit long in this PR. Was trying to make it obvious what the function is supposed to do. I'm now doing something similar - returning directly instead of storing it in a variable and returning the variable.

Reply:
> Was trying to make it obvious what the function is supposed to do.

I believe that should be doable without long variable names.

Reply:
Done.